virtcontainers/kata_agent.go

```diff
@@ -742,12 +745,38 @@ func (k *kataAgent) createContainer(sandbox *Sandbox, c *Container) (p *Process,
 		return nil, err
 	}
 
+	var devInfos []config.DeviceInfo
+	for _, devInfo := range c.config.DeviceInfos {
+		if devInfo.DevType == "e" {
+			filename := devInfo.ContainerPath
+			epheStorage := &grpc.Storage{
+				Driver:     "ephemeral",
+				Source:     "tmpfs",
+				Fstype:     "tmpfs",
+				MountPoint: filename,
```

Comment:
IIUC you need "/ephemeralPath/" + filepath.Base(devInfo.ContainerPath) here, to match line 744 below?

Reply:
Thanks for pointing that out. Turns out I was looping over a wrong slice. Instead of looping over `ociSpec.Mounts` I was looping over `newMounts`. The reason it worked during testing because I kept the source path intact, which meant it created a path like, `/var/lib/kubelet/pods/366c3a75-4869-11e8-b479-507b9ddd5ce4/volumes/kubernetes.io~empty-dir/cache-volume` inside a VM. 

I will update the PR with the fix.